exercises/grains/src/test/java/GrainsTest.java

```diff
@@ -0,0 +1,108 @@
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.math.BigInteger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class GrainsTest {
+
+    private Grains grains;
+
+    @Before
+    public void setup() {
+        grains = new Grains();
+    }
+
+
+    @Test
+    public void countAtSquare1() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(1);
+        assertEquals(new BigInteger("1"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void countAtSquare2() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(2);
+        assertEquals(new BigInteger("2"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void countAtSquare3() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(3);
+        assertEquals(new BigInteger("4"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void countAtSquare4() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(4);
+        assertEquals(new BigInteger("8"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void countAtSquare16() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(16);
+        assertEquals(new BigInteger("32768"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void countAtSquare32() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(32);
+        assertEquals(new BigInteger("2147483648"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void countAtSquare64() {
+        BigInteger result = grains.computeNumberOfGrainsOnSquare(64);
+        assertEquals(new BigInteger("9223372036854775808"), result);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void errorOnNullBoardSize() {
+        try {
+            grains.computeNumberOfGrainsOnSquare(0);
```

Comment:
Please use [ExpectedException](https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html). You can look at the [phone-number tests](https://github.com/exercism/java/blob/a5c8c2144a25b13feeb884cf6e1eeb71097b09ac/exercises/phone-number/src/test/java/PhoneNumberTest.java) for an example of how it's used 🙂

Reply:
I read about it in the notes and somehow thought it was to test exception generation, as I do in my personal project. I should have look for examples in the code.
I am updating the code right now.

Reply:
Yes, I read about it and somehow confused it with the custom exception.
By the way, this was removed in JUnit5, in favour of a new method `Assertions.assertThrows`. But that's a topic for another time.

Reply:
Good to know! It's probably worth us updating the exercises to use that at some point then 🙂